app/src/main/java/ca/ualberta/CMPUT3012019T02/alexandria/model/message/LocationMessage.java

```diff
@@ -6,18 +6,22 @@
 
 public class LocationMessage extends Message {
 
-    public LocationMessage(String content, String status, Date date, String sender) {
-        super("Location", content, status, date, sender);
+    public LocationMessage(Location content, String status, Date date, String sender) {
+        if (!(content instanceof Location.class)) {
+            throw new IllegalArgumentException("location content needs to be a Location object");
+        }
+
+        super("location", content, status, date, sender);
     }
 
     public Location getLocation() {
-        // TODO Finish implementation
-        throw new UnsupportedOperationException();
+        return this.content;
     }
 
     public void setLocation(Location location) {
-        // TODO Finish implementation
-        throw new UnsupportedOperationException();
+        if (!(content instanceof Location.class)) {
+            throw new IllegalArgumentException("location content needs to be a Location object");
+        }
+        this.content = location;
```

Comment:
You cannot convert directly. You need to serialized, but we don't need this functionality for part 3

Reply:
You could do something like getLong+","+getLAt
And the other way could be something like getContent().split(",")
But we don't necessarily need to implement for project part 3

Reply:
okay, I'll change this to a string check for now.

I am curious though, why do we need to serialize the location?